File: 구명보트.py
def solution(people, limit):
    answer = 0
    left = 0
    right = len(people)-1
    people.sort()

    while left<=right:
        if people[right] + people[left]<=limit:
            right -=1
            left +=1
            answer+=1
        elif left == right:
            answer+=1
            break
        else:
            answer +=1
            right -=1
    return answer

File: 구명보트.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        int left = 0;
        int right = people.length-1;
        Arrays.sort(people);
        while(left<=right){
            if (people[right]+people[left] <= limit) {
                answer++;
                right--;
                left++;
            }
            else if(right == left){
                answer ++;
                break;
            }
            else{
                right--;
                answer++;
            }
        }
        return answer;
    }
}
